src/redux/data.js

```diff
@@ -541,13 +551,30 @@ const watchPendingTransactions = (
   }
 };
 
+export const addNewSubscriber = (subscriber, type) => (dispatch, getState) => {
+  const { subscribers } = getState().data;
+  const newSubscribers = {
+    ...subscribers,
+  };
+
+  newSubscribers[type].push(subscriber);
```

Comment:
Technically the push breaks redux immutable update pattern (I presume it still works tho?)

Reply:
ok this is probs overkill so can ignore, but i think we can do:
const newSubscribers = { ...subscribers };
newSubscribers[type] = concat(newSubscribers[type], subscriber)

that way it is immutable + leaves the rest of the objects just the same without cloning all the things (since we only care about updating one of the items)

Reply:
https://github.com/rainbow-me/rainbow/pull/997